pkg/sources/filesystem/filesystem.go

```diff
@@ -68,9 +68,10 @@ func (s *Source) Init(aCtx context.Context, name string, jobId, sourceId int64,
 	s.verify = verify
 
 	var conn sourcespb.Filesystem
-	err := anypb.UnmarshalTo(connection, &conn, proto.UnmarshalOptions{})
-	if err != nil {
-		errors.WrapPrefix(err, "error unmarshalling connection", 0)
+	if err := anypb.UnmarshalTo(connection, &conn, proto.UnmarshalOptions{}); err != nil {
+		if err := errors.WrapPrefix(err, "error unmarshalling connection", 0); err != nil {
+			return err
+		}
```

Comment:
**Suggestion:** While this if statement is not technically wrong, I think we can always return `errors.WrapPrefix` here.